pkg/controller/master/upgrade/job_controller_test.go

```diff
@@ -115,6 +115,20 @@ func TestJobHandler_OnChanged(t *testing.T) {
 				err:     nil,
 			},
 		},
+		{
+			name: "a previous upgrading manifest job should not complete upgrade",
+			given: input{
+				key:     testJobName,
+				job:     newTestChartJobBuilder().Completed().Build(),
+				plan:    newTestPlanBuilder().Build(),
+				upgrade: newTestUpgradeBuilder().WithLabel(harvesterLatestUpgradeLabel, "false").ChartUpgradeStatus(v1.ConditionUnknown, "", "").Build(),
+			},
+			expected: output{
+				job:     newTestChartJobBuilder().Completed().Build(),
+				upgrade: newTestUpgradeBuilder().WithLabel(harvesterLatestUpgradeLabel, "false").ChartUpgradeStatus(v1.ConditionUnknown, "", "").Build(),
```

Comment:
Do we want to change it to something like the following?

```suggestion
				job: newJobBuilder(testJobName).
					WithLabel(harvesterUpgradeLabel, "test-upgrade-old").
					WithLabel(harvesterUpgradeComponentLabel, manifestComponent).
					Completed().Build(),
				plan:    newTestPlanBuilder().Build(),
				upgrade: newTestUpgradeBuilder().WithLabel(harvesterLatestUpgradeLabel, "true").ChartUpgradeStatus(v1.ConditionUnknown, "", "").Build(),
			},
			expected: output{
				job: newJobBuilder(testJobName).
					WithLabel(harvesterUpgradeLabel, "test-upgrade-old").
					WithLabel(harvesterUpgradeComponentLabel, manifestComponent).
					Completed().Build(),
				upgrade: newTestUpgradeBuilder().WithLabel(harvesterLatestUpgradeLabel, "true").ChartUpgradeStatus(v1.ConditionUnknown, "", "").Build(),
```

Reply:
Thanks, fixed!